### DISCUSSION TITLE
Inquiry About the Motor Driver in Crazyflie 2.1+

### DISCUSSION THREAD

**Discussion by bluesungho** [bitcraze/discussions#1711] [Q&A]
Dear Bitcraze AB

Hello,
I am a graduate student currently conducting research on mini drones. Recently, I purchased the Crazyflie 2.1+ for research purposes, and I would like to ask a few technical questions regarding its motor driver design.

Regarding the motor driver in the Crazyflie 2.1+, I have the following questions:

It appears that the Crazyflie 2.1+ uses coreless 720 motors and CJ2302 FET components.

What were the criteria for selecting these components?
Could you briefly explain the configuration and control method of the motor driver?
What is the role of pins P7, P8, P9, P10, and what does CONN_2 signify in the motor driver section?

Is the motor driver in the Crazyflie 2.1+ a commercially available driver, or is it custom-designed?

If it is a commercial driver, could you let me know which product was used?
If it is a custom-designed driver, how were the components selected for its design?
For reference, I am currently conducting experiments using coreless 8520 motors.

I have tried using DRV8833 and DRV8411 drivers, but encountered issues where the voltage was not delivered properly to the driver.
When using an ESC 5A, the voltage delivery improved (over 80%), but the low resolution of the ESC (90Hz) made precise control difficult.
I hope to improve these issues by referencing the design and control methods of the Crazyflie 2.1+ motor driver. Your response would be greatly appreciated and would significantly aid my research.

Thank you for your time and assistance.

Best regards,
Seongho Choi
Kyonggi University, Department of Mechanical Systems Engineering

**Comment by tobbeanton** [comment]
All these questions are a bit out of scope for what we have time to handle. We have a description of the the HW design for the old Crazyflie 1.0 you can find [here](https://wiki.bitcraze.io/projects:crazyflie:hardware:explained). A lot of the basics for CF1.0 are the same so you can learn a lot from it. Other then that you will be able to find a lot of good information on the internet.
